### PR TITLE
Authorise Role Assignment Service

### PR DESCRIPTION
**Pull request by hemantt**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by MSancaktutar — CHANGES_REQUESTED** [review]
Could you enable on ITHC and Perftest, as well please?

**Review by MSancaktutar — COMMENTED** [review]
Sorry, but if they need to access case details, this is not the correct env var. In any case, general s2s authorisation will be (also) required.

File: k8s/aat/common/ccd/data-store-api.yaml [hunk 1]
```diff
@@ -25,7 +25,8 @@ spec:
         CCD_DM_DOMAIN: ^https?://(?:api-gateway\.preprod\.dm\.reform\.hmcts\.net|dm-store-aat\.service\.core-compute-aat\.internal(?::\d+)?)
         CCD_CONDITIONAL_APIS_CASE_ASSIGNED_USER_AND_ROLES_ENABLED: true
         CCD_S2S_AUTHORISED_SERVICES_CASE_USER_ROLES: aac_manage_case_assignment,fpl_case_service
-        DUMMY_RESTART_VAR: 1
+        DATA_STORE_S2S_AUTHORISED_SERVICES: am_role_assignment_service
```
Comment (smathangi) [inline k8s/aat/common/ccd/data-store-api.yaml:28]: It just override chart's env variable. So, you might need to include all other values from here https://github.com/hmcts/ccd-data-store-api/blob/master/charts/ccd-data-store-api/values.yaml#L34

File: k8s/aat/common/ccd/data-store-api.yaml [hunk 2]
```diff
@@ -25,7 +25,8 @@ spec:
...
+        DUMMY_RESTART_VAR: 0
```
Comment (smathangi) [inline k8s/aat/common/ccd/data-store-api.yaml:29]: delete these dummy variables

**Review by MSancaktutar — CHANGES_REQUESTED** [review]
This will deny access to all current allowed list configured in chart config.
Please **make sure you copy the full list from respective configs** and append am_role_assignment_service.

**Pull request closed without merging**